### ISSUE TITLE
Pre-submission Enquiry

### ISSUE THREAD

**Issue by BenWinchester** [openjournals/joss#1123]
Dear Editorial Board,

My enquiry is how continuous development relates to JOSS submissions.

I work on [CLOVER](https://github.com/CLOVER-energy/CLOVER/tree/master), a research tool for simulating the energy needs of minigrids with a focus on rural electrification. As part of our development process, we have multiple researchers each developing different aspects of the code:
* New means of generating electricity, which is carried out on the [`dev`](https://github.com/CLOVER-energy/CLOVER/tree/dev) branch;
* More advanced means of modelling diesel generators, which is carried out on the [`diesel-dev`](https://github.com/CLOVER-energy/CLOVER/tree/diesel-dev) branch.

As each of these features is released, we plan to issue a new version. (Tags `v5.1.0` and `v5.2.0` are currently reserved for this.) I was wondering how this relates to JOSS submissions. I.E., would we (provided we meet the three-month, 1000-line etc. eligibility criteria) qualify for a JOSS submission each time we release such a version of the code? We are wondering when we should prepare a JOSS submission as we are concerned that, by submitting to JOSS, we would miss out on the opportunity of submitting, both the two features mentioned above, and future features.

Many Thanks,
[Ben Winchester](https://github.com/BenWinchester)

**Comment by arfon** [comment]
Hi @BenWinchester. Thanks for the question. It's a good one, and a little complex to answer. While JOSS does in theory support this kind of publication model (multiple papers for different aspects of a long-lived project), what you are describing may raise concerns by our editorial team of 'salami slicing' of work (i.e., multiple papers in a short period of time about the same project) which would be considered unethical.

What we have done in the past is had a second (not sure if there has been a third yet) JOSS publication for a piece of software upon a _major_ release e.g., going from v1.0 to v2.0.

I'm not sure if I've helped with this comment or not? :-)

**Comment by BenWinchester** [comment]
Hi @arfon, that makes sense, thank you.

We're not aiming to salami slice, more to reflect that there are several PhD and MSc projects ongoing which are working on developing the code, most likely over the next year or two, and we didn't want these contributions to go unrecognised. We've now made a submission to JOSS as our software stands and will potentially submit again once we've got these features into the code, likely in a year or two!

Thanks again :smile:
